plenario/api/validator.py

```diff
@@ -0,0 +1,173 @@
+    'shape': lambda x: get_shape_table(x)
+}
+
+
+def get_shape_table(shtable_name):
+    """A simple helper method to get a table that corresponds to a ShapeMetadata
+    record. This exists because while MetaTable.point_table exists, ShapeMetadata
+    doesn't have an analogous function.
+
+    :param shtable_name: name of shape table
+
+    :returns: table instance corresponding to ShapeMetadata record"""
+
+    return session.query(ShapeMetadata)\
+        .filter(ShapeMetadata.dataset_name == shtable_name)\
+        .first()\
+        .shape_table
+
+
+def convert(request_args):
+    """Convert a dictionary of arguments from strings to their types. How the
+    values are converted are specified by the converters dictionary defined
+    above.
+
+    :param request_args: dictionary of request arguments
+
+    :returns: converted dictionary"""
+
+    for key, value in request_args.items():
+        try:
+            request_args[key] = converters[key](value)
+        except:
+            # Failed transactions, which we do expect, can cause
+            # a DatabaseError with Postgres. Failing to rollback
+            # prevents further queries from being carried out.
+            session.rollback()
+
+
+def validate(validator_cls, request_args, defaults=True):
+    """Validate a dictionary of arguments. Substitute all missing fields with
+    defaults if not explicitly told to do otherwise.
+
+    :param validator_cls: what kind of validator to use
+    :param request_args: dictionary of arguments from a request object
+    :param defaults: if false, only substitute default values for provided args
+
+    :returns: ValidatorResult namedtuple"""
+
+    if defaults:
+        # this validator will return results/defaults for all fields
+        validator = validator_cls()
+    else:
+        # this validator will return results/defaults for only the fields
+        # provided in the request
+        validator = validator_cls(only=request_args.keys())
+
+    # convert string values to corresponding types so the validator can work
+    convert(request_args)
+    # returns validated parameters as strings
+    result = validator.dump(request_args)
+    # convert strings back to correct types
+    convert(result.data)
+
+    # determine unchecked parameters provided in the request
+    unchecked = set(request_args.keys()) - set(validator.fields.keys())
+
+    # determine if a dataset was provided to grab the column names
+    columns = []
+    if 'dataset_name' in request_args:
+        # this little bit of result formatting helps make things easier, but I
+        # eventually need to find a better place for it
+        result.data['metatable'] = result.data['dataset']
+        result.data['dataset'] = result.data['dataset'].point_table
+        columns += result.data['dataset'].columns.keys()
+    if 'shape' in request_args:
+        columns += result.data['shape'].columns.keys()
```

Comment:
We want to kill this. Don't let users rely on shape columns being accessible through the dunder syntax.